.github/pull_request_template.md

```diff
@@ -1,32 +1,39 @@
 <!--
 
-Describe in detail the changes you are proposing, and the rationale.
+** Thank you for your contribution! Please read this carefully! **
 
-See the contributing guide:
-
-https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md
+Please make sure you go through the checklist below. If your PR does not meet all requirements, please file it
+as a draft PR. Core team members will only review your PR once it meets all the requirements below (unless your
+change is something as trivial as a typo fix).
 
 -->
 
-<!--
+<!-- If your PR resolves an issue, please add it here. -->
+Resolves # 
 
-Link all GitHub issues fixed by this PR, and add references to prior related PRs.
-Make sure to first open an issue, get community approval and only then create Pull Request to resolve it.
-All Pull Requests must have an issue attached to them
+## Target Release
 
--->
+1.7.0
 
-Resolves #
+## Checklist
 
-## Target Release
+<!-- This checklist is mandatory for all PRs: -->
 
-<!--
+- [ ] I have read the [contribution guide](https://github.com/opentofu/opentofu/blob/main/CONTRIBUTING.md).
+- [ ] I have not used an AI coding assistant to create this PR.
+- [ ] I have marked any code I have not written myself (including modified code, e.g. copied from other places and then modified) with a comment indicating where it came from. 
 
-In normal circumstances we only target changes at the upcoming minor
-release, or as a patch to the current minor version. If you need to
-port a security fix to an older release, highlight this here by listing
-all targeted releases.
+### Go checklist
 
--->
+<-- If your PR contains Go code, please make sure you check off all items on this list: --> 
 
-1.7.0
+- [ ] I have run golangci-lint on my change and receive no errors relevant to my code.
+- [ ] I have added tests for all relevant use cases of my code.
```

Comment:
IIRC, there are some tools that can report the unit test coverage based on the code submitted in a PR.

Reply:
We had a requirement for it in here, but we took it out because we don't want to mandate a specific code coverage. It should really cover the important code paths.